socs/agents/ld_monitor/agent.py

```diff
@@ -0,0 +1,424 @@
+                return self.data_dict
+
+            # "status" sentence
+            elif param == 'ST':
+                faultcode = int(self.data.split(',')[-1].split('*')[0], 16)
+                data_split = [int(i) for i in self.data.split(',')[1:-1]]
+
+                self.newdata_dict = {
+                    'd_type': 3,
+                    'alarm_r': data_split[0],
+                    'alarm_o': data_split[1],
+                    'alarm_y': data_split[2],
+                    'delay_g': data_split[3],
+                    'clear': data_split[4],
+                    'r_timer': data_split[5],
+                    'o_timer': data_split[6],
+                    'y_timer': data_split[7],
+                    'g_timer': data_split[8],
+                    'allclear_timer': data_split[9],
+                    'faultcode': faultcode
+                }
+
+                self.data_dict.update(self.newdata_dict)
+                return self.data_dict
+
+            # disregard "alarm timers" sentence but update sentence type
+            elif param == 'WT':
+                self.newdata_dict = {'d_type': 4}
+                self.data_dict.update(self.newdata_dict)
+                return self.data_dict
+
+    def read_cycle(self):
+        """
+        In each cycle data is read and then parsed following
+        the format required to publish data to the ocs feed
+        """
+        try:
+            cycle_data = {}
+            self.read_data()
+
+            # updates time since last strike if previous strike data exists
+            if self.data_dict['time_last'] == -1.:
+                self.data_dict['tsince_last'] = -1.
+            else:
+                self.data_dict['tsince_last'] = (time.time()
+                                                 - self.data_dict['time_last'])
+
+            # parse data to ocs agent feed format
+            for key in self.data_dict:
+                cycle_data[key] = {'value': self.data_dict[key]}
+
+            if self.verbose:
+                print(cycle_data)
+            return cycle_data
+
+        except BaseException:
+            pass
+            if self.verbose:
+                print('Passing to next data iteration')
+
+
+class ld_monitorAgent:
+    """Monitor the Lightning Detector data via UDP.
+
+    Parameters
+    ----------
+    agent : OCSAgent
+        OCSAgent object which forms this Agent
+    unit : int
+    sample_interval : float
+        Time between samples in seconds.
+
+    Attributes
+    ----------
+    agent : OCSAgent
+        OCSAgent object which forms this Agent
+    take_data : bool
+        Tracks whether or not the agent is actively issuing SNMP GET commands
+        to the ibootbar. Setting to false stops sending commands.
+    log : txaio.tx.Logger
+        txaio logger object, created by the OCSAgent
+    """
+
+    def __init__(self, agent, unit=1, sample_interval=15.):
+
+        self.unit = unit

```

Comment:
This doesn't get used anywhere. I was also wondering what it was and was going to ask for a description in the docstring. It seems to maybe correspond to some address for a given device to monitor?

Since it's not used, I'd say either drop it, or implement the use of it. If you implement it, please add a description to the docstring.